### PR TITLE
Replace ke::Vector with std::vector.

### PR DESCRIPTION
**Pull request by dvander**

### CODE REVIEW

File: core/MenuStyle_Base.cpp [hunk 1]
```diff
@@ -661,16 +661,16 @@ bool CBaseMenu::InsertItem(unsigned int position, const char *info, const ItemDr
...
 }
 
 bool CBaseMenu::RemoveItem(unsigned int position)
 {
-	if (position >= m_items.length())
+	if (position >= m_items.size())
 		return false;
 
-	m_items.remove(position);
+	ke::RemoveAt(&m_items, position);
```
Comment (Headline) [inline core/MenuStyle_Base.cpp:673]: Any reason we’re not doing

`m_items.erase(m_items.begin()+ position)`

File: core/logic/CDataPack.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ size_t CDataPack::CreateMemory(size_t size, void **addr)
 	val.type = CDataPackType::Raw;
 	val.pData.vval = new uint8_t[size + sizeof(size)];
 	reinterpret_cast<size_t *>(val.pData.vval)[0] = size;
-	elements.insert(position, val);
+	ke::InsertAt(&elements, position, val);
```
Comment (Headline) [inline core/logic/CDataPack.cpp:90]: std::vector::insert?

File: core/logic/DatabaseConfBuilder.h [hunk 3]
```diff
@@ -90,7 +90,7 @@ class ConfDbInfoList : public ke::Vector<ConfDbInfo *>
 		m_DefDriver = std::string(input);
 	}
 	void ReleaseMembers() {
-		for (size_t i = 0; i < this->length(); i++) {
+		for (size_t i = 0; i < this->size(); i++) {
```
Comment (Headline) [inline core/logic/DatabaseConfBuilder.h:93]: range based for

Comment (dvander) [reply core/logic/DatabaseConfBuilder.h:93]: To minimize friction (and time spent writing these patches), I tried to make these changes as syntax-preserving as I could. Improvements like range-based for loops can be done as needed. The argument for RemoveAt/InsertAt is weaker though so I'll revert those.

**Pull request merged**